app/utils.py

```diff
@@ -0,0 +1,34 @@
+from simple_salesforce import Salesforce
+from . import (
+    SF_USERNAME,
+    SF_PASSWORD,
+    SF_SECURITY_TOKEN,
+)
+
+
+def fetch_cases(comma_separated_case_numbers):
+    sf = Salesforce(username=SF_USERNAME, password=SF_PASSWORD, security_token=SF_SECURITY_TOKEN, domain='test')
+    case_numbers = comma_separated_case_numbers.split(',')
+    case_numbers_querystring = ','.join(repr(str(num)) for num in case_numbers) if case_numbers else "''"
+    cases = sf.query(
+        'SELECT id, Contract__c, Description, CaseNumber from Case WHERE CaseNumber IN ({})'
+        .format(case_numbers_querystring))['records']
+
+    contract_ids = [c['Contract__c'] for c in cases]
+    contract_ids_querystring = ','.join(repr(str(id)) for id in contract_ids)
```

Comment:
Why are yo putting id in `repr(str(id))`? Does Salesforce support id in SQL query as integers? or is it works only for strings ? same question for `case numbers`

Reply:
ids must be put between single quotes, so repr acts as a shortcut to achieve that (maybe it's not semantically correct)

Reply:
Also if you want you can put 2 `for` statements in one line for list comprehention :) https://stackoverflow.com/questions/18551458/how-to-frame-two-for-loops-in-list-comprehension-python

Reply:
> ids must be put between single quotes, so repr acts as a shortcut to achieve that (maybe it's not semantically correct)

Great! good catch